core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -166,12 +172,15 @@ protected final void deframeFailed(Throwable cause) {
    * @param trailers the received trailers
    * @param status the status extracted from the trailers
    */
-  protected void inboundTrailersReceived(Metadata.Trailers trailers, Status status) {
-    Preconditions.checkNotNull(trailers, "trailers");
+  protected final void inboundTrailersReceived(Metadata.Trailers trailers, Status status) {
+    checkNotNull(status);
+    checkNotNull(trailers);
     if (inboundPhase() == Phase.STATUS) {
-      log.log(Level.INFO, "Received trailers on closed stream {0}\n {1}\n {3}",
+      log.log(Level.INFO, "Received trailers on closed stream {0}\n {1}\n {2}",
           new Object[]{id(), status, trailers});
     }
+    checkState(status == null, "status has already been set");
```

Comment:
This doesn't look right. It seems guaranteed to fail. Shouldn't it be `this.status`? Ditto on the next line.